### ISSUE TITLE
Markdown: Syntax highlighting sometimes removed

### ISSUE THREAD

**Issue by SimenB** [prettier/prettier#3176]
**Prettier 1.8.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBGAdAAgGIQE4DWmMAFnJnnAA4SYCGUAJpmJXfJgJ4QCuemAIzwMwJTADM8EALaYABtLoBneHjnoAOlC2ZdAeSoJicPNICWUOgBtMACjjoA5lgAqJ85asAaTGbemfAHEzGEwAIWUxfExg0IBlMisrAEp6JmJOQyQdXTk8pRIczEcQlitochIYGColJAB6epLSHgF0SGl6gB5uPgB9HiUTS2k4AD56gCs4FSKwZmnZqF1i0tE4MAJeUIBaAUxpTj6hEULl3LyigDkIeGzzzAAlais6MHI5Ht48AaG8EfGckwAHcQmJevxYgAJVqYQbDOijLRaADMWEePGWck4dH+vigKmsVnURT0ywA6hZGBBgXUii5aBZCUlMABtACauKgAF1bFUanVGjj-lQCM4OvUEPVqWAlPUmTAiQBiUFMGlKHaKgSpaCYSlq2lcXgHOicTBQOBwZgwWjU4FQcp0ZhwMEmc0QRgOSZKTDRADCJAgkFe8E4XSEmHqY2RD3k+TOKy4XPxzKsRTjcnTDJYZE2Rr4mAAbiYlGZdRBxJhOXiGM6CXxyCEAOQ+hVEq35lgMDJZdN5OQFdPC5Y7HbFvCl6B9vIgLwgCBUGBlgnIUDKGDIcTWIZzk6bOAwOJUN4WRzIGB4HhwOcyzfb68gb0ADzCwn3h8RcAAMhY4HerDu848DUwEAEz-oBx4TiYyAgIohB2lAs4gFQeAWDAlKMKQyAABwAAxzkM5jnpeD6llAjhWHAACKPC3H+SBbgBD5aph2FIKBc4XnQZhWKefoyIosFQBUyHwi4dACEoEFwAAvrJQA)
```sh
--parser markdown
```

**Input:**
````markdown
1. Fork the repo and create your branch from `master`.

   Open terminal (e.g. Terminal, iTerm, Git Bash or Git Shell) and type:
   ```sh
   git clone https://github.com/<your_username>/jest
   cd jest
   git checkout -b my_branch
   ```
   Note:
   Replace `<your_username>` with your GitHub username

3. Run `yarn install`.
   On Windows:
   To install [Yarn](https://yarnpkg.com/en/docs/install#windows-tab) on Windows you may need to download either node.js or Chocolatey<br />

    ```sh
    yarn install
    ```
    To check your version of Yarn and ensure it's installed you can type:
    ```sh
    yarn --version
    ```
````

**Output:**
````markdown
1. Fork the repo and create your branch from `master`.

   Open terminal (e.g. Terminal, iTerm, Git Bash or Git Shell) and type:

   ```sh
   git clone https://github.com/<your_username>/jest
   cd jest
   git checkout -b my_branch
   ```

   Note: Replace `<your_username>` with your GitHub username

2. Run `yarn install`. On Windows: To install
   [Yarn](https://yarnpkg.com/en/docs/install#windows-tab) on Windows you may
   need to download either node.js or Chocolatey<br />

       yarn install

   To check your version of Yarn and ensure it's installed you can type:

       yarn --version

````

**Expected behavior:**
Quite big example, but I expect point 3 to be formatted same as point 1.

Currently, the <code>```sh</code> part is removed

**Comment by azz** [comment]
Looks like it's due to the extra space.

**Prettier 1.8.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBGAdAAgDpU-gAyIGcALXfTATwEMAnPASymJhoBt2LCjv8AVCJjCk4YANbUIAVzqYAbnDrFG0TBABmmAJr08NKABNMCYrLiZGMAOTFLLNpzjGqM4QcwwqABzhI+mEQEZAG0DJgAtBGKyqpQAUEgADQgEN4wccTIoDSsyBocxHApAEZ0NBJwMADK3hXMAObIMHTSxSCGEGD5he0AVsQAHgBC5ZU1NAC2cAAyzHA97EUpMjDe0jAATIvLIHXKSsggk-TinQDuUMl7dMwwAOqMhjCkyAAcAAwpRZOMza3tFRQBrsOAARWkEHgO3abBKj2eryQmxSLRojHYjQAwhBJicjlBoAsUtIivwaCUskgCks4ABfOlAA)
```sh
--parser markdown
```

**Input:**
````markdown
1. 
   ```sh
   yarn install
   ```
    To check your version of Yarn and ensure it's installed you can type:
    ```sh
    yarn --version
    ```
````

**Output:**
````markdown
1. ```sh
   yarn install
   ```
   To check your version of Yarn and ensure it's installed you can type:
       yarn --version

````

**Comment by azz** [comment]
This has been fixed in `1.8.1`.
